app/controllers/article_controller.rb

```diff
@@ -1,89 +1,66 @@
 class ArticleController < ApplicationController
-  before_action :authorized?, :except => [:list, :show, :tooltip, :graph]
-
-
   include AbstractXmlController
 
-  DEFAULT_ARTICLES_PER_GRAPH = 40
-  GIS_DECIMAL_PRECISION = 5
-  LIST_NUM_COLUMNS = 3
-
-  def authorized?
-    unless user_signed_in?
-      redirect_to dashboard_path
-    end
-  end
+  before_action :authorized?, except: [:list, :show, :tooltip, :graph]
 
   def tooltip
     render partial: 'tooltip'
   end
 
   def list
-    @categories = @collection.categories
-    # Differences from previous implementation:
-    # 1. List of articles needs to be collection-specific
-    # 2. List should be displayed within the category treeview
-    # 3. Uncategorized articles should be listed below
-    if @collection.is_a?(DocumentSet)
-      @uncategorized_articles = @collection.articles.joins('LEFT JOIN articles_categories ac ON articles.id = ac.article_id').where('ac.category_id IS NULL').reorder(:title)
-    else
-      @uncategorized_articles = @collection.articles.where.not(:id => @collection.articles.joins(:categories).pluck(:id)).reorder(:title)
-    end
+    articles = @collection.articles.includes(:categories)
+    @categories = @collection.categories.includes(:articles)
 
-    # Create a 2D array of articles/subjects for each category
-    # where the articles are sorted alphabetically vertically
-    # instead of horizontally -- so, going down each column
-    @vertical_articles = {} # Holds articles for each category
+    @vertical_articles = {}
     @categories.each do |category|
-      articles = category.articles_list(@collection) # The articles for this category
-      @vertical_articles[category] = sort_vertically(articles)
+      current_articles = articles.where(categories: { id: category.id })
+      @vertical_articles[category] = sort_vertically(current_articles)
     end
 
-    # Do the same for uncategorized articles
-    if @uncategorized_articles.present?
-      @uncategorized_articles = sort_vertically(@uncategorized_articles)
-    end
+    @uncategorized_articles = sort_vertically(
+      articles.where(categories: { id: nil }).reorder(:title)
+    )
   end
 
   def delete
-    if @article.link_list.empty? && @article.target_article_links.empty?
-      if @article.created_by_id == current_user.id || current_user.like_owner?(@collection)
-        @article.destroy 
-        redirect_to collection_subjects_path(@collection.owner, @collection)
-      else
-        flash.alert = t('.only_subject_owner_can_delete')
-        redirect_to collection_article_show_path(@collection.owner, @collection, @article.id)
-      end
+    result = Article::Destroy.call(
```

Comment:
I'm confused by this -- this seems to skp a lot of the checks on links to the article being destroyed.  How does this new syntax address these?

Reply:
I moved the checks to interactors/article/destroy.rb. It contains the same logic.

Reply:
Oh, interesting!